exonum-java-binding-core/src/main/java/com/exonum/binding/transaction/InternalTransactionContext.java

```diff
@@ -0,0 +1,38 @@
+package com.exonum.binding.transaction;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.storage.database.Fork;
+
+public class InternalTransactionContext implements TransactionContext {
+  private final Fork fork;
+  private final HashCode hash;
+  private final PublicKey authorPk;
+
+  /**
+   * Creates Internal Transaction Context.
+   * @param fork fork
+   * @param hash hash
+   * @param authorPk author private key
+   */
+  public InternalTransactionContext(Fork fork, HashCode hash, PublicKey authorPk) {
+    this.fork = fork;
+    this.hash = hash;
+    this.authorPk = authorPk;
+  }
+
+  @Override
+  public Fork getFork() {
+    return fork;
+  }
+
+  @Override
+  public HashCode getTransactionMessageHash() {
+    return hash;
+  }
+
+  @Override
+  public PublicKey getAuthorPk() {
+    return authorPk;
+  }
```

Comment:
toString()?

Reply:
Are you asking if there must be one :upside_down_face: ?

Reply:
yeap